src/utils/compare.ts

```diff
@@ -2,11 +2,16 @@ import { hexToRgb } from "./convert";
 
 import { HSL, HSV, RGB } from "../types";
 
+interface ColorCompare {
+  [key: string]: number;
+}
+
 export const equalColorObjects = (first: HSL | HSV | RGB, second: HSL | HSV | RGB): boolean => {
   if (first === second) return true;
 
   for (const prop in first) {
-    if (first[prop] !== second[prop]) return false;
+    if (((first as unknown) as ColorCompare)[prop] !== ((second as unknown) as ColorCompare)[prop])
```

Comment:
Uh, looks brutal, but as far as I understand there are no other options.

Reply:
Yeah, sorry.

I'll keep looking around, maybe there's a better solution I'm just not thinking of right now. 

Iterating over object keys is always going to be a bit awkward with TS though.

Reply:
We can merge this solution, but could I ask you to add a comment here describing what're we doing here and why?

Reply:
Absolutely. I can also remove the interface declaration and just in-line as `Record<string, number>` now they I think about it.